frontend/components/static-badge-maker.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+function escapeField(s) {
+  return encodeURIComponent(s.replace(/-/g, '--').replace(/_/g, '__'));
+}
+
+export default class StaticBadgeMaker extends React.Component {
+  makeBadgeUri () {
+    let url = `${this.props.baseUri}/badge/`;
+    url += escapeField(this.subjectInput.value);
+    url += '-' + escapeField(this.statusInput.value);
+    url += '-' + escapeField(this.colorInput.value);
+    url += '.svg';
+    return url;
+  }
+
+  handleSubmit (e) {
+    e.preventDefault();
+    document.location = this.makeBadgeUri();
```

Comment:
What about popping up a new tab rather than setting the location of the current tab?